packages/core-flows/src/definition/promotion/create-campaigns.ts

```diff
@@ -0,0 +1,13 @@
+import { CampaignDTO, CreateCampaignDTO } from "@medusajs/types"
+import { createWorkflow } from "@medusajs/workflows-sdk"
+import { createCampaignsStep } from "../../handlers/promotion"
+
+type WorkflowInput = { campaignsData: CreateCampaignDTO[] }
+type WorkflowOutput = CampaignDTO[]
+
+export const createCampaignsWorkflow = createWorkflow<
+  WorkflowInput,
+  WorkflowOutput
+>("create-campaigns", (input) => {
```

Comment:
todo: If we can avoid these explicit typings, I think we should

cc @adrien2p 
```suggestion
export const createCampaignsWorkflow = createWorkflow("create-campaigns", (input) => {
```

Reply:
Any reason why this should be removed? I still need to access `input.campaignsData` to pass to the `createCampaignsStep`. 

Example of what it looks like without
![Screenshot 2024-01-22 at 08 04 51](https://github.com/medusajs/medusa/assets/5105988/0b3d2969-9720-4490-811b-690e6223f46c)

Reply:
you can if you type the input as being `WorkflowData<WorkflowInput`

Reply:
cc @olivermrbl

Reply:
you can do the same for the output, either a WorkflowData or WorkflowDataProperties of something

Reply:
Ah, I see. 🤦🏻 Thanks!